shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.mm

```diff
@@ -0,0 +1,93 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterMacOSGLCompositor.h"
+
+#import <OpenGL/gl.h>
+#import "flutter/fml/logging.h"
+#import "flutter/fml/platform/darwin/cf_utils.h"
+#import "flutter/shell/platform/darwin/macos/framework/Source/MacOSSwitchableGLContext.h"
+#import "third_party/skia/include/core/SkCanvas.h"
+#import "third_party/skia/include/core/SkSurface.h"
+#import "third_party/skia/include/gpu/gl/GrGLAssembleInterface.h"
+#import "third_party/skia/include/utils/mac/SkCGUtils.h"
+
+#include <unistd.h>
+
+namespace flutter {
+
+FlutterMacOSGLCompositor::FlutterMacOSGLCompositor(FlutterViewController* view_controller,
+                                                   NSOpenGLContext* open_gl_context)
+    : view_controller_(view_controller) {
+  openGLContext_ = [[NSOpenGLContext alloc] initWithFormat:open_gl_context.pixelFormat
```

Comment:
why do we need to create a new one here? Why not save the ptr?

Reply:
Done